pkg/gofr/gofr.go

```diff
@@ -94,9 +95,39 @@ func New() *App {
 
 	app.subscriptionManager = newSubscriptionManager(app.container)
 
+	// static fileserver
+	currentWd, _ := os.Getwd()
+	checkDirectory := filepath.Join(currentWd, "public")
+
+	if _, err := os.Stat(checkDirectory); !os.IsNotExist(err) {
+		app.AddStaticFiles("public", checkDirectory)
```

Comment:
I'm a bit worried about the security concern here.

It's somehow logic to serve a public folder, but let's consider this feature is added to all existing repository.

So they are likely to serve files that were present in a public folder that weren't planned to be served.

I would expect people who does want to share public file to look for AddStaticFiles in documentation and add it in their code.

Maybe, it's OK  if the code is merged as this, it should come with a big warning or breaking change mention in the release.

Reply:
Hi @ccoVeille,
Someone has mentioned in the issue as well to serve public folder if it exists. So that was the reason behind me adding that logic.

Reply:
Then as I said, it should be mentioned in the changelog/release in a very clear way

Also, I would expect to see a log entry about either serving public folder (so here around this if) or better at the end of "AddStaticFiles" method

Reply:
> Someone has mentioned in the issue as well to serve public folder if it exists. So that was the reason behind me adding that logic.

For reference:

>We can follow a convention also. If a public folder is present, files in that folder can be accessed by the server. 
 
_Originally posted by @vikash in https://github.com/gofr-dev/gofr/issues/502#issuecomment-2121842503_  

It's not because someone asks for something that a good idea for security.

I'm sure @vikash (hi :wave:) who is also the issue's author won't have any issue with having a public folder being served automatically. But other people wouldn't notice the change and might face security issue.

I'm raising my concern. As I said, I'm OK with having the feature of a public folder being automagically served.